src/test/java/com/conveyal/datatools/HandleCorruptGTFSFileTest.java

```diff
@@ -0,0 +1,39 @@
+package com.conveyal.datatools;
+
+import com.conveyal.datatools.manager.models.FeedSource;
+import com.conveyal.datatools.manager.models.FeedVersion;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static com.conveyal.datatools.TestUtils.createFeedVersionFromGtfsZip;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Run test to handle a corrupt GTFS file gracefully.
+ */
+public class HandleCorruptGTFSFileTest {
+    private static FeedSource mockFeedSource;
+
+    @BeforeAll
+    public static void setUp() throws IOException {
+        // start server if it isn't already running
+        DatatoolsTest.setUp();
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        mockFeedSource.delete();
+    }
+
+    @Test
+    public void canHandleCorruptGTFSFile() {
+        mockFeedSource = new FeedSource("Corrupt");
+        Persistence.feedSources.create(mockFeedSource);
+        FeedVersion corruptFeedVersion = createFeedVersionFromGtfsZip(mockFeedSource, "corrupt-gtfs-file.zip");
+        assertEquals(corruptFeedVersion.feedLoadResult.fatalException, "java.util.zip.ZipException: error in opening zip file");
```

Comment:
It might be good to build in some other assertions here. E.g., we probably want to check that the ValidateFeedJob never kicked off and that there is no remnants of the feed version in the postgres database following the ProcessSingleFeedJob run.

Reply:
I've made a update to address to the sub jobs. Still not sure on postgrres remnats.